packages/mask/src/plugins/Trader/apis/1inch/index.ts

```diff
@@ -0,0 +1,27 @@
+import BigNumber from 'bignumber.js'
+import { first } from 'lodash-unified'
+import { BIPS_BASE, ONE_INCH_BASE_URL } from '../../constants'
+import type { SwapOneErrorResponse, SwapQuoteOneResponse, SwapQuoteOneRequest } from '../../types'
+import urlcat from 'urlcat'
+
+import { getNetworkTypeFromChainId, NetworkType } from '@masknet/web3-shared-evm'
+import { fetchJSON } from '../../../../../../web3-providers/src/helpers'
+
+export async function swapOneQuote(request: SwapQuoteOneRequest) {
+    const params: Record<string, string | number> = Object.fromEntries(Object.entries(request))
```

Comment:
why

Reply:
@developerfred You can do shallow clone with
```suggestion
    const params = { ...request }
```

But I don't think you really need cloning. Following line should not mix in this function, it should be moved outside, that means you should make the the parameters satisfy this function instead.

```ts
    if (request.slippage) params.slippage = new BigNumber(request.slippage).dividedBy(BIPS_BASE).toFixed()
```